python-package/lightgbm/basic.py

```diff
@@ -1840,7 +1883,17 @@ def set_init_score(self, init_score):
         """
         self.init_score = init_score
         if self.handle is not None and init_score is not None:
-            init_score = list_to_1d_numpy(init_score, np.float64, name='init_score')
+            if is_1d_collection(init_score):
+                init_score = list_to_1d_numpy(init_score, np.float64, name='init_score')
+            elif is_2d_collection(init_score):
+                init_score = data_to_2d_numpy(init_score, np.float64, name='init_score')
+                init_score = init_score.ravel()
```

Comment:
I'm not sure that simple `ravel()` can be used here. I guess we should change the order (`ravel(order='F')`).
> For multi-class task, the y_pred is group by class_id first, then group by row_id. If you want to get i-th row y_pred in j-th class, the access way is y_pred[j * num_data + i] and you should group grad and hess in this way as well.

https://github.com/microsoft/LightGBM/blob/237ac299fcffa2b6ed9c68917dfdf0767e88e59c/python-package/lightgbm/basic.py#L1182-L1187